test/e2e/tests/custom_stages.py

```diff
@@ -0,0 +1,57 @@
+NUM_STAGES = 4
+
+def get_stages():
+    stages = []
+    for i in range(1, NUM_STAGES + 1):
+        stages.append("stage_" + str(i))
+    return stages
+
+def get_init_args():    
+    return {
+        "kwargs": {
+            "project": "custom_stages",
+            "stages": get_stages(),
+        }
+    }
+
+def get_jobs():
+    jobs = []
+    for i in range(1, NUM_STAGES + 1):
+        job = {
+            "kwargs": {
+                "command": "sleep 2",
+                "ci-stage": "stage_" + str(i),
+                "description": str(i),
+                "pipeline": "siesta",
+            }
+        }
+        jobs.append(job)
+    return jobs
+
+def get_run_build_args():
+    return {}
+
+def check_run(run):
```

Comment:
Nice, thanks a lot for this! Please ensure that there's a newline at the end of the file